compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2935,24 +2935,19 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
         ctx1.gadt.addToConstraint(typeHoles)
         ctx1
 
-    val qctx1 = QuotesImpl()(using ctx1)
-
-    val matcher = new Matcher.QuoteMatcher[qctx1.type](qctx1) {
-      def patternHoleSymbol: qctx1.reflect.Symbol = dotc.core.Symbols.defn.QuotedRuntimePatterns_patternHole.asInstanceOf
-      def higherOrderHoleSymbol: qctx1.reflect.Symbol = dotc.core.Symbols.defn.QuotedRuntimePatterns_higherOrderHole.asInstanceOf
-    }
-
     val matchings =
-      if pat1.isType then matcher.termMatch(scrutinee.asInstanceOf[matcher.qctx.reflect.Term], pat1.asInstanceOf[matcher.qctx.reflect.Term])
-      else matcher.termMatch(scrutinee.asInstanceOf[matcher.qctx.reflect.Term], pat1.asInstanceOf[matcher.qctx.reflect.Term])
+      if pat1.isType then QuoteMatcher.termMatch(scrutinee, pat1)(using ctx1)
+      else QuoteMatcher.termMatch(scrutinee, pat1)(using ctx1)
```

Comment:
This `if/else` seems not useful -- both branches have the same code.

Reply:
Removed unnecessary duplication